go.mod

```diff
@@ -1,28 +1,40 @@
 module github.com/iotexproject/go-pkgs
 
-go 1.17
+go 1.21
+
+toolchain go1.21.4
 
 require (
 	github.com/cespare/cp v1.1.1 // indirect
-	github.com/deckarep/golang-set v1.7.1 // indirect
 	github.com/dustinxie/gmsm v1.4.0
-	github.com/ethereum/go-ethereum v1.10.4
+	github.com/ethereum/go-ethereum v1.13.5-0.20240410090245-9dcf8aae4742
```

Comment:
I used the lastest commit in the master branch for testing, since geth v1.13.15 hasn't been released yet.

Reply:
For prod usage, we might wait until geth v1.13.15 is released

Reply:
can we just use v1.13.12 (which is the Cancun release version we use for iotex-core)
Does v1.13.12 fix this memory issue?

Reply:
no. pls check the geth's pr on the top